### PR TITLE
Improve debug formatting

### PR DESCRIPTION
**Pull request by andybalaam**

### CODE REVIEW

File: src/csr.rs [hunk 1]
```diff
@@ -91,6 +89,22 @@ macro_rules! implement_csr {
                 Self::new()
             }
         }
+        impl core::fmt::Debug for $struct_name {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                f.write_fmt(format_args!(
+                    "{} {{ registers: {{{}}} }}",
```
Comment (Johnabell) [inline src/csr.rs]:
```suggestion
                    "{} {{ registers: {{ {} }} }}",
```

**Review by Johnabell — COMMENTED** [review]
Nice work

File: src/csr.rs [hunk 2]
```diff
@@ -132,6 +146,28 @@ implement_csr!(CSR64, i64);
...
     use super::*;
 
+    #[test]
+    fn debug_format_32() {
+        let csr_32 = CSR32::default();
+        csr_32.read_write(34, 43);
+        csr_32.read_write(85, 58);
+        assert_eq!(
+            format!("{:?}", csr_32),
+            "CSR32 { registers: {34: 43, 85: 58} }"
```
Comment (Johnabell) [inline src/csr.rs]:
```suggestion
            "CSR32 { registers: { 34: 43, 85: 58 } }"
```

File: src/csr.rs [hunk 3]
```diff
@@ -132,6 +146,28 @@ implement_csr!(CSR64, i64);
...
+    }
+
+    #[test]
+    fn debug_format_64() {
+        let csr_64 = CSR64::default();
+        csr_64.read_write(14, 41);
+        csr_64.read_write(85, 58);
+        assert_eq!(
+            format!("{:?}", csr_64),
+            "CSR64 { registers: {14: 41, 85: 58} }"
```
Comment (Johnabell) [inline src/csr.rs]:
```suggestion
            "CSR64 { registers: { 14: 41, 85: 58 } }"
```

File: src/csr.rs (hunk 1 above)
Comment (andybalaam) [reply src/csr.rs]: Done in 60fc2a8

**Review by Johnabell — APPROVED** [review]
Nice thanks

**Pull request merged**